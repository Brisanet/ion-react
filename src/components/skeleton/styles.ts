import { css, keyframes, styled } from 'styled-components';

const shimmer = keyframes`
  75% {
    transform: translateX(100%);
  }
  100% {
    opacity: 0;
  }
`;

export const Skeleton = styled.span`
  ${({ theme }) => css`
    display: block;
    position: relative;
    overflow: hidden;
    background-color: ${theme.colors.neutral[1]};

    &::after {
      content: '';
      position: absolute;
      inset: 0;
      transform: translateX(-100%);
      background: linear-gradient(
        90deg,
        rgba(246, 247, 248, 0.2) 0%,
        rgba(155, 160, 177, 0.2) 48.52%,
        rgba(246, 247, 248, 0.2) 100%
      );
      animation: ${shimmer} 1.5s infinite ease-in-out;
    }
  `}
`;
