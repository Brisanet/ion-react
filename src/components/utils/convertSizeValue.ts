/**
 * Get a size value based on its type.
 * @param {string | number} value - The size value.
 * @returns The value itself if it's a string, or the value in pixels if it is a number.
 **/
export const convertSizeValue = (value: string | number): string =>
  typeof value === 'number' ? `${value}px` : value;
