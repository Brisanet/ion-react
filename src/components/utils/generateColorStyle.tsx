// Tipo para objeto de estilos  da função geradora de estilos
type ColorStyle = {
    background: string;
    color: string;
    border?: string;
  };
  
  /* 
    Função que gera cores de Background, Border e cor de
    fonte a fim de reduzir repetições de codigo
  
    Seguir o padrão de parâmetros: 
    1- Cor do background;
    2- Cor da fonte do botão;
    3- (Opicional) Cor da borda;
  */
 export function generateColorStyle(
    bgColor: string,
    fontColor: string,
    borderColor: string
  ) {
    let styles: ColorStyle = {
      background: bgColor,
      color: fontColor,
    };
  
    borderColor && (styles.border = `1px solid ${borderColor}`);
  
    return styles;
  }