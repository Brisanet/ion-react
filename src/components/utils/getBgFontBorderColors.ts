interface ObjStyles {
  background: string;
  color: string;
  border?: string;
}

/**
 * Function to prevent repetition of style blocks.
 *
 * @param background - Background color
 * @param color - The font color
 * @param border - Border attribute styles - example: ´1px solid blue´
 * @returns A Javascript object containing color properties.
 */
export function getBgFontBorderColors({
  background,
  color,
  border = '',
}: ObjStyles): ObjStyles {
  return {
    background,
    color,
    border,
  };
}
