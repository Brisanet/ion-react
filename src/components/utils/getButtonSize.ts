/**
 * Function to prevent repetition of style blocks.
 *
 * @param buttonSize - Height and min-width
 * @param padding - Button padding
 * @param fontSize - The size of font
 * @returns An Javascript object containing size properties.
 */
export function getButtonSize(
  buttonSize: string,
  padding: string,
  fontSize: string
) {
  let size = {
    height: buttonSize,
    minWidth: buttonSize,
    padding: padding,
    fontSize: fontSize,
  };

  return size;
}
