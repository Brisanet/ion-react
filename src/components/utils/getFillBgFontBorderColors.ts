type ColorStyleResponse = {
  border: string;
  backgroundColor: string;
  color: string;
  svg: {
    fill: string;
  };
};

interface StyleProps {
  svgFillColor: string;
  backgroundColor: string;
  fontColor: string;
  borderColor: string;
}

/**
 * Function to prevent repetition of style blocks.
 *
 * @param svgFillColor - Fill color to be used in SVG
 * @param backgroundColor - Background color
 * @param fontColor - The font color
 * @param borderColor - Border color
 * @returns A Javascript object containing style properties.
 */
export function getFillBgFontBorderColors({
  backgroundColor,
  fontColor,
  borderColor,
  svgFillColor,
}: StyleProps) {
  let styles: ColorStyleResponse = {
    backgroundColor: backgroundColor,
    color: fontColor,
    border: `1px solid ${borderColor}`,
    svg: {
      fill: svgFillColor,
    },
  };

  return styles;
}
