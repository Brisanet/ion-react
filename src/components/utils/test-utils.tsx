import theme from '@ion/styles/theme';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
