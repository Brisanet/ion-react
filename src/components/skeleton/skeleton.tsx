import { Skeleton } from './styles';

export interface IonSkeletonProps {
  width?: string | number;
  height?: string | number;
  radius?: string | number;
}

const convertToStyle = (value: string | number) => {
  return typeof value === 'number' ? `${value}px` : value;
};

export const IonSkeleton = ({
  height = 100,
  radius = 0,
  width = 100,
}: IonSkeletonProps) => {
  const widthStyle = convertToStyle(width);
  const heightStyle = convertToStyle(height);
  const radiusStyle = convertToStyle(radius);

  return (
    <Skeleton
      style={{
        width: widthStyle,
        height: heightStyle,
        borderRadius: radiusStyle,
      }}
    ></Skeleton>
  );
};
