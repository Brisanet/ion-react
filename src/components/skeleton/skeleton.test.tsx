import { renderWithTheme } from '../utils/test-utils';
import { IonSkeleton, IonSkeletonProps } from './skeleton';

const sut = (props?: IonSkeletonProps) => {
  return renderWithTheme(<IonSkeleton {...props} />);
};

describe('IonSkeleton', () => {
  it('should render default skeleton', () => {
    const { container } = sut();
    expect(container.firstChild).toHaveStyle({
      width: '100px',
      height: '100px',
      borderRadius: '0px',
    });
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render skeleton with correct unit when width and height are numbers', () => {
    const sizes = { width: 200, height: 200, radius: 50 };
    const { container } = sut({ ...sizes });
    expect(container.firstChild).toHaveStyle({
      width: `${sizes.width}px`,
      height: `${sizes.height}px`,
      borderRadius: `${sizes.radius}px`,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render skeleton with correct unit when width and height are strings', () => {
    const sizes = { width: '20rem', height: '200px', radius: '50%' };
    const { container } = sut({ ...sizes });
    expect(container.firstChild).toHaveStyle({
      width: sizes.width,
      height: sizes.height,
      borderRadius: sizes.radius,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
