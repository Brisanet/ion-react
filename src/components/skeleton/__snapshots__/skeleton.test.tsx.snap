// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IonSkeleton should render default skeleton 1`] = `
.c0 {
  display: block;
  position: relative;
  overflow: hidden;
  background-color: #fcfcfd;
}

.c0::after {
  content: '';
  position: absolute;
  inset: 0;
  transform: translateX(-100%);
  background: linear-gradient(         90deg,         rgba(246, 247, 248, 0.2) 0%,         rgba(155, 160, 177, 0.2) 48.52%,         rgba(246, 247, 248, 0.2) 100%       );
  animation: cLHmAQ 1.5s infinite ease-in-out;
}

<span
  class="c0"
  style="width: 100px; height: 100px; border-radius: 0px;"
/>
`;

exports[`IonSkeleton should render skeleton with correct unit when width and height are numbers 1`] = `
.c0 {
  display: block;
  position: relative;
  overflow: hidden;
  background-color: #fcfcfd;
}

.c0::after {
  content: '';
  position: absolute;
  inset: 0;
  transform: translateX(-100%);
  background: linear-gradient(         90deg,         rgba(246, 247, 248, 0.2) 0%,         rgba(155, 160, 177, 0.2) 48.52%,         rgba(246, 247, 248, 0.2) 100%       );
  animation: cLHmAQ 1.5s infinite ease-in-out;
}

<span
  class="c0"
  style="width: 200px; height: 200px; border-radius: 50px;"
/>
`;

exports[`IonSkeleton should render skeleton with correct unit when width and height are strings 1`] = `
.c0 {
  display: block;
  position: relative;
  overflow: hidden;
  background-color: #fcfcfd;
}

.c0::after {
  content: '';
  position: absolute;
  inset: 0;
  transform: translateX(-100%);
  background: linear-gradient(         90deg,         rgba(246, 247, 248, 0.2) 0%,         rgba(155, 160, 177, 0.2) 48.52%,         rgba(246, 247, 248, 0.2) 100%       );
  animation: cLHmAQ 1.5s infinite ease-in-out;
}

<span
  class="c0"
  style="width: 20rem; height: 200px; border-radius: 50%;"
/>
`;
